src/style.js

```diff
@@ -1,24 +1,12 @@
 import { Component } from 'react'
-import StyleSheetRegistry from './stylesheet-registry'
-
-const styleSheetRegistry = new StyleSheetRegistry()
+import RegistryContext from './stylesheet-registry-context'
 
 export default class JSXStyle extends Component {
   constructor(props) {
     super(props)
     this.prevProps = {}
   }
 
-  static dynamic(info) {
```

Comment:
Please revert this. It computes dynamic classes at runtime. https://github.com/zeit/styled-jsx/blob/00fcb7bcad9c8af91167056f93c63858d24769ba/src/_utils.js#L347

Reply:
In fact dynamic styles won't work with your solutions. Before we were relying on the singleton and using it in this static method. Now the method needs to be instance aware and therefore can't be static.

This is a problem because at build time (when we add `JSXStyle.dynamic` calls) we don't know anything about instances.

We'll need to think about a solution.

Reply:
probably you can move these outside of the class 

https://github.com/zeit/styled-jsx/blob/00fcb7bcad9c8af91167056f93c63858d24769ba/src/stylesheet-registry.js#L129-L195

make them IIFEs (the cache can be shared by all the instances of the registry I think) and export `computeId` so that `JSXStyle` can use it

Reply:
I wondered if `dynamic` was used by the Babel compiler, but I couldn't find a reference to it anywhere. This is another place where tests are obviously lacking.

Reply:
yeah we don't have e2e tests because we need to set up some infra